### ISSUE TITLE
Updating from 3.1.34.3 to 3.4.0.2 migrations fail with syntax error

### ISSUE THREAD

**Issue by rmales** [craftcms/cms#5527]
### Description

When updating from 3.1.34.3 to 3.4.0.2 I get the following error from the migrations.

```
Applying new migrations ... error: The command "'/var/www/html/craft' 'migrate/all' '--no-content'" failed.

Exit Code: 1(General error)

Working directory: /var/www/html

Output:
================


Error Output:
================
Exception 'yii\base\InvalidArgumentException' with message 'Syntax error.'

in /var/www/html/vendor/yiisoft/yii2/helpers/BaseJson.php:133

Stack trace:
#0 /var/www/html/vendor/yiisoft/yii2/helpers/BaseJson.php(107): yii\helpers\BaseJson::handleJsonError(4)
#1 /var/www/html/vendor/craftcms/cms/src/services/ProjectConfig.php(1718): yii\helpers\BaseJson::decode('a:7:{s:12:"date...')
#2 /var/www/html/vendor/craftcms/cms/src/services/ProjectConfig.php(1604): craft\services\ProjectConfig->_loadInternalConfigData()
#3 /var/www/html/vendor/craftcms/cms/src/services/ProjectConfig.php(1593): craft\services\ProjectConfig->_getStoredConfig()
#4 /var/www/html/vendor/craftcms/cms/src/services/ProjectConfig.php(397): craft\services\ProjectConfig->_getLoadedConfig()
#5 /var/www/html/vendor/craftcms/cms/src/base/ApplicationTrait.php(1375): craft\services\ProjectConfig->get('system.timeZone')
#6 /var/www/html/vendor/craftcms/cms/src/base/ApplicationTrait.php(1333): craft\console\Application->_setTimeZone()
#7 /var/www/html/vendor/craftcms/cms/src/console/Application.php(57): craft\console\Application->_preInit()
#8 /var/www/html/vendor/yiisoft/yii2/base/BaseObject.php(109): craft\console\Application->init()
#9 /var/www/html/vendor/yiisoft/yii2/base/Application.php(206): yii\base\BaseObject->__construct(Array)
#10 /var/www/html/vendor/yiisoft/yii2/console/Application.php(89): yii\base\Application->__construct(Array)
#11 /var/www/html/vendor/craftcms/cms/src/console/Application.php(48): yii\console\Application->__construct(Array)
#12 [internal function]: craft\console\Application->__construct(Array)
#13 /var/www/html/vendor/yiisoft/yii2/di/Container.php(400): ReflectionClass->newInstanceArgs(Array)
#14 /var/www/html/vendor/yiisoft/yii2/di/Container.php(159): yii\di\Container->build('craft\\console\\A...', Array, Array)
#15 /var/www/html/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get('craft\\console\\A...', Array, Array)
#16 /var/www/html/vendor/craftcms/cms/bootstrap/bootstrap.php(245): yii\BaseYii::createObject(Array)
#17 /var/www/html/vendor/craftcms/cms/bootstrap/console.php(43): require('/var/www/html/v...')
#18 /var/www/html/craft(21): require('/var/www/html/v...')
#19 {main}


Output:


```

### Steps to reproduce

1. Install 3.1.34.3
2. `./craft update all`

### Additional info

- Craft version: 3.1.34.3
- PHP version: 7.2.19
- Database driver & version: MySQL 5.6.44
- Plugins & versions: None

**Comment by rmales** [comment]
Closing as I think this may have been due to not having `mysqldump` available in our PHP container. Sorry for the spurious report.

**Comment by brandonkelly** [comment]
Hm, looks like a real Craft bug to me. Any chance you can send a database backup from **before** you updated to Craft 3.4, plus your `composer.json` and `composer.lock` files, over to support@craftcms.com?

**Comment by kerrig** [comment]
I'm also seeing this issue but updating from 3.1.25.

### Additional info
Craft version: 3.1.25
PHP version: 7.1.30
Database driver & version: MySQL 5.6.39

I will also send an email to Craft support with what @brandonkelly requested.

**Comment by brandonkelly** [comment]
Craft 3.4.2 is out now with the fix for this.

**Comment by rmales** [comment]
Thanks. This upgrade does also seem to fail if mysqldump is not available, even when you decline to create a database backup. Is this expected or should I open another issue?

```
Backup the database? (yes|no) [yes]:no
Skipping database backup.
Performing update with Composer ... done
Applying new migrations ... error: The command "'/var/www/html/craft' 'migrate/all' '--no-content'" failed.

Exit Code: 1(General error)

Working directory: /var/www/html

Output:
================
Yii Migration Tool (based on Yii v2.0.32)



Error Output:
================
Error backing up the database: The shell command "mysqldump --defaults-extra-file="/var/www/html/storage/runtime/temp/my.cnf" --add-drop-table --comments --create-options --dump-date --no-autocommit --routines --default-character-set=utf8 --set-charset --triggers --single-transaction --no-data --result-file="/var/www/html/storage/backups/test_200201_180149_xg120imiby_v3.4.2.sql" craft && mysqldump --defaults-extra-file="/var/www/html/storage/runtime/temp/my.cnf" --add-drop-table --comments --create-options --dump-date --no-autocommit --routines --default-character-set=utf8 --set-charset --triggers --no-create-info --ignore-table=craft.assetindexdata --ignore-table=craft.assettransformindex --ignore-table=craft.sessions --ignore-table=craft.templatecaches --ignore-table=craft.templatecachequeries --ignore-table=craft.templatecacheelements --ignore-table=craft.cache --ignore-table=craft.templatecachecriteria craft >> "/var/www/html/storage/backups/test_200201_180149_xg120imiby_v3.4.2.sql"" failed with exit code 127: sh: 1: mysqldump: not found


Output:

Yii Migration Tool (based on Yii v2.0.32)


```

**Comment by brandonkelly** [comment]
@rmales In the future a separate GH issue would be preferred, yes, since it’s unrelated to the main issue here. No big deal this time though – just fixed that error via 0ec898c79afa002df52fbfe765ecbc210fe8d16d.
